crates/web5_cli/src/doctor.rs

```diff
@@ -0,0 +1,218 @@
+use std::collections::HashMap;
+
+use crate::CheckType;
+
+pub struct HealthCheckState {
+    cli_version: Option<Option<String>>,
+    dependencies: Option<HashMap<String, bool>>,
+    env_vars: Option<HashMap<String, bool>>,
+    connectivity: Option<bool>,
+    basic_functionality: Option<bool>,
+}
+
+impl HealthCheckState {
+    fn new() -> Self {
+        Self {
+            cli_version: None,
+            dependencies: None,
+            env_vars: None,
+            connectivity: None,
+            basic_functionality: None,
+        }
+    }
+}
+
+fn check_cli_version() -> Option<Option<String>> {
+    Some(Some(env!("CARGO_PKG_VERSION").to_string())) // This will return the binary version.
+}
+
+fn check_dependencies() -> Option<HashMap<String, bool>> {
+    // TODO : Implement this function
+    // DOUBT  : Are we expecting to check system dependencies or dependencies in Cargo.toml ?
+    // If cargo dependencies then how exactly shall we check the versions ?
+    None
+}
+
+fn check_environment_variables() -> Option<HashMap<String, bool>> {
+    let mut env_vars = HashMap::new();
+    let vars = vec!["PORTABLE_DID"]; // Add env_vars that you want to include in health checkup
+
+    for var in vars {
+        let status = std::env::var(var).is_ok();
+        env_vars.insert(var.to_string(), status);
+    }
+
+    Some(env_vars)
+}
+
+async fn check_connectivity() -> Option<bool> {
```

Comment:
why do we need to check connectivity here?

for did web and did dht resolution ?

Reply:
Yes for web mainly
`web5 did create web https://blackgirlbytes.com`